oblib/parser.py

```diff
@@ -124,64 +155,126 @@ def from_JSON_string(self, json_string, entrypoint_name=None):
 
         json_string (str): String containing JSON
         entrypoint_name (str): Optional name of the entrypoint.
-
-        TODO: Currently this method works in some cases but testing is incomplete
         """
 
+        # Create a validation error which can be used to maintain a list of error messages
+        validation_errors = ValidationErrors("Error(s) found in input JSON")
```

Comment:
I think this counterproductive, `validation_errors` will never be empty and thus will always be raised

Reply:
Its not always raised.  it only is raised if errors are actually added

Reply:
the if condition is len(validation_errors) > 0, and we initialize validation_errors with this message, so we always hit the raise statement

Reply:
You have to call append() at least once in order to add a ValidationError to ValidaitonErrors and cause the raise condition to occur.

Reply:
You are correct, I see that.  Thanks for the patience.